lib/resourcedelete/batch.go

```diff
@@ -37,3 +37,32 @@ func DeleteJobv1(ctx context.Context, client batchclientv1.JobsGetter, required
 	}
 	return true, nil
 }
+
+// DeleteCronJobv1 checks the given resource for a valid delete annotation. If found
+// it checks the status of a previously issued delete request. If delete has not been
+// requested and in UpdatingMode it will issue a delete request.
+func DeleteCronJobv1(ctx context.Context, client batchclientv1.CronJobsGetter, required *batchv1.CronJob,
+	updateMode bool) (bool, error) {
+
+	if delAnnoFound, err := ValidDeleteAnnotation(required.Annotations); !delAnnoFound || err != nil {
+		return delAnnoFound, err
+	}
+	existing, err := client.CronJobs(required.Namespace).Get(ctx, required.Name, metav1.GetOptions{})
+	resource := Resource{
+		Kind:      "cronjob",
+		Namespace: required.Namespace,
+		Name:      required.Name,
+	}
+	if deleteRequested, err := GetDeleteProgress(resource, err); err == nil {
```

Comment:
It'd read a bit better if the condition was inverse:

```go
deleteRequested, err := GetDeleteProgress(resource, err)
if err != nil {
   return true, fmt.Errorf("Error running delete for %s, err=%v", resource, err)
}
if ... {
  if ... {
    return ...
  }
  SetDeleteRequested(...
}
return true, nil
```

Reply:
Me too! We would return early rather than nest deeper.

I'll change the condition in this current PR, and in the future, we can refactor the other conditions in the `resourcedelete` package (or using the DRY principle we will leave reuse this condition).